File: Arise.FileSyncer.Service/Ipc/Messages/SetAllowPairingMessage.cs
using Arise.FileSyncer.Core;

namespace Arise.FileSyncer.Service.Ipc.Messages
{
    public class SetAllowPairingMessage : IpcMessage
    {
        public override string Command => "setAllowPairing";

        public bool AllowPairing { get; set; }

        internal override void Process(IpcController ipc)
        {
            Log.Info("Setting 'AllowPairing' to " + AllowPairing);
            ipc.Service.Peer.AllowPairing = AllowPairing;

            if (AllowPairing)
            {
                ipc.Service.Discovery.SendDiscoveryMessage();
            }
        }
    }
}

File: Arise.FileSyncer.ElectronUI/src/main/ipc/messages/setAllowPairingMessage.ts
import { IpcMessage } from "../ipcMessage";

export const Command = "setAllowPairing";

export interface Message extends IpcMessage {
    AllowPairing: boolean,
}

export function Create(allow: boolean): Message {
    let message: Message = {
        Command: Command,
        AllowPairing: allow,
    };

    return message;
}
